p4-16/psa/PSA.mdk

```diff
@@ -2066,10 +2068,53 @@ to select a member from the group, and the action specification
 corresponding to that member is applied. The dynamic selection algorithm is
 specified as a parameter when instantiating the action selector.
 
-Action selector members may only specify action types defined
-in the `actions` attribute of the implemented table. All actions in a group must be
-of the same type. The action parameters for actions in the same group are allowed
-to differ, and the action of different groups in a selector may be different.
+Action selector members may only specify action types defined in the
+`actions` attribute of the implemented table.  PSA implementations are
+allowed to restrict all actions in a group to have the same action
+name, but they may allow more general configurations, i.e. different
+action names among members of the same group.  If a PSA implementation
+does have this restriction, it must still allow different members of
+the same group to have different action parameter values.  PSA
+implementations must also allow the action names of different groups
+in a selector to be different.
```

Comment:
I think this paragraph may benefit from a bit of reorganization for clarity.
In particular, it would be good to first list the requirements: must have at lease one action in each group, allow different actions in different groups. And introduce the `psa_emtpy_group_action` property (it simply occurs later) and its semantics: provide the must have action when there are no entries in the group. Then discuss the possible extensions and restrictions: may allow different parameter values, may restrict to one action.

Reply:
Thanks for detailed review comments.  Take a look at the latest version when you get a chance.  I have attempted to address all of your comments, although I am only replying to this one.